syntax = "proto3";

message GuidanceData {
  float deltaT = x;
  float errVel = 1; // PID inputs
  float errDrift = 2;

  float errVelI = 3;
  float errVelD = 4;
  
  float kP_vel = 5; // PID constants for velocity PID
  float kI_vel = 6;
  float kD_vel = 7;

  float velP = x;
  float velI = x;
  float velD = x;

  float kP_drift = 8; // PID constants for drift PID
  float kI_drift = 9;
  float kD_drift = 10;

  float leftOutputVel = 11; // left motor output from velocity PID
  float rightOutputVel = 12; // right motor output from velocity PID
  float leftOutputDrift = 13; // left motor output from drift PID
  float rightOutputDrift = 14; // right motor output from drift PID
  
  float leftTotalPID = x; // combined left motor output from vel and drift PIDs
  float rightTotalPID = x; // combined left motor output from vel and drift PIDs

  float leftPower = 15; // combined left motor output from vel and drift PIDs
  float rightPower = 16; // combined right motor output from vel and drift PIDs
  float propPower = 17; // combined right motor output from vel and drift PIDs

  uint32 segNum = 18; // current arena segment number
}
